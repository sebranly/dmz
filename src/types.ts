// TODO: remove
export enum Template {
  Typescript = 'typescript',
  Javascript = 'javascript'
}

export type TemplateLanguage = {
  language: Template;
};
