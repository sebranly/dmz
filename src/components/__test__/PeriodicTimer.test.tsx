import * as React from 'react';
import { render } from '@testing-library/react';
import { PeriodicTimer } from '../PeriodicTimer';
import { APITime, TimeFrequency, TimeStatus, TimeType } from '../../types';

test('PeriodicTimer', () => {
  const time: APITime = {
    type: TimeType.Challenges,
    name: 'Daily Challenges',
    frequency: TimeFrequency.Daily,
    status: TimeStatus.Reset,
    time: 1678078800
  };

  const createComponent = (currentTimestamp: number, time: APITime) => (
    <PeriodicTimer currentTimestamp={currentTimestamp} time={time} />
  );

  const { container: containerDaily } = render(createComponent(1678078801, time));
  expect(containerDaily.childNodes).toMatchSnapshot();

  time.name = 'Weekly Challenges';
  time.frequency = TimeFrequency.Weekly;

  const { container: containerWeekly } = render(createComponent(1678078801, time));
  expect(containerWeekly.childNodes).toMatchSnapshot();
});
