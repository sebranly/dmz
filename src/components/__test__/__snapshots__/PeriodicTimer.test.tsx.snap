// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodicTimer 1`] = `
NodeList [
  <div
    class="bg-neutral-800 border-solid border-2 rounded-lg border-blue-500 m-2.5 p-2.5 relative w-72 sm:w-80"
  >
    <div
      class="font-bold my-1 text-lg text-blue-500"
    >
      Daily Challenges
    </div>
    <div
      class="font-bold my-1 text-lg"
    >
      It resets in
    </div>
    <ul
      class="timer-card flex justify-center"
    >
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-5xl"
        >
          23
        </span>
        <div
          class="text-sm"
        >
          hours
        </div>
      </li>
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-5xl"
        >
          59
        </span>
        <div
          class="text-sm"
        >
          minutes
        </div>
      </li>
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-5xl"
        >
          59
        </span>
        <div
          class="text-sm"
        >
          seconds
        </div>
      </li>
    </ul>
    <div
      class="text-sm"
    >
      <div
        class="flex text-left pl-2.5"
      >
        <div
          class="grow"
        >
          Daily Reset:
        </div>
        <div
          class="font-bold pr-1 text-blue-500"
        >
          06:00 AM
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`PeriodicTimer 2`] = `
NodeList [
  <div
    class="bg-neutral-800 border-solid border-2 rounded-lg border-blue-500 m-2.5 p-2.5 relative w-72 sm:w-80"
  >
    <div
      class="font-bold my-1 text-lg text-blue-500"
    >
      Weekly Challenges
    </div>
    <div
      class="font-bold my-1 text-lg"
    >
      It resets in
    </div>
    <ul
      class="timer-card flex justify-center"
    >
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-4xl"
        >
          06
        </span>
        <div
          class="text-sm"
        >
          days
        </div>
      </li>
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-4xl"
        >
          23
        </span>
        <div
          class="text-sm"
        >
          hrs
        </div>
      </li>
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-4xl"
        >
          59
        </span>
        <div
          class="text-sm"
        >
          min
        </div>
      </li>
      <li
        class="inline-block mx-2.5"
      >
        <span
          class="font-bold text-4xl"
        >
          59
        </span>
        <div
          class="text-sm"
        >
          sec
        </div>
      </li>
    </ul>
    <div
      class="text-sm"
    >
      <div
        class="flex text-left pl-2.5"
      >
        <div
          class="grow"
        >
          Weekly Reset:
        </div>
        <div
          class="font-bold pr-1 text-blue-500"
        >
          Mon 06:00 AM
        </div>
      </div>
    </div>
  </div>,
]
`;
